test/rbi/rewriters/filter_versions_test.rb

```diff
@@ -0,0 +1,284 @@
+
+        # @version > 0.4.0
+        class Baz; end
+
+        # @version > 0.3.0
+        class Buzz; end
+      RBI
+
+      tree = Parser.parse_string(rbi)
+
+      Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.4.0"))
+      assert_equal(<<~RBI, tree.string)
+        # @version > 0.4.0-prerelease
+        class Bar; end
+
+        # @version > 0.3.0
+        class Buzz; end
+      RBI
+    end
+
+    def test_filter_versions_equals
+      rbi = <<~RBI
+        # @version = 1.0.0
+        class Foo; end
+
+        # @version = 0.4.0-prerelease
+        class Bar; end
+
+        # @version = 0.4.0
+        class Baz; end
+
+        # @version = 0.3.0
+        class Buzz; end
+      RBI
+
+      tree = Parser.parse_string(rbi)
+
+      Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.4.0"))
+      assert_equal(<<~RBI, tree.string)
+        # @version = 0.4.0
+        class Baz; end
+      RBI
+    end
+
+    def test_filter_versions_greater_than_or_equals
+      rbi = <<~RBI
+        # @version >= 1.0.0
+        class Foo; end
+
+        # @version >= 0.4.0-prerelease
+        class Bar; end
+
+        # @version >= 0.4.0
+        class Baz; end
+
+        # @version >= 0.3.0
+        class Buzz; end
+      RBI
+
+      tree = Parser.parse_string(rbi)
+
+      Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.4.0"))
+      assert_equal(<<~RBI, tree.string)
+        # @version >= 0.4.0-prerelease
+        class Bar; end
+
+        # @version >= 0.4.0
+        class Baz; end
+
+        # @version >= 0.3.0
+        class Buzz; end
+      RBI
+    end
+
+    def test_filter_versions_less_than_or_equals
+      rbi = <<~RBI
+        # @version <= 1.0.0
+        class Foo; end
+
+        # @version <= 0.4.0-prerelease
+        class Bar; end
+
+        # @version <= 0.4.0
+        class Baz; end
+
+        # @version <= 0.3.0
+        class Buzz; end
+      RBI
+
+      tree = Parser.parse_string(rbi)
+
+      Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.4.0"))
+      assert_equal(<<~RBI, tree.string)
+        # @version <= 1.0.0
+        class Foo; end
+
+        # @version <= 0.4.0
+        class Baz; end
+      RBI
+    end
+
+    def test_filter_versions_prerelease
+      rbi = <<~RBI
+        # @version <= 1.0.0
+        class Foo; end
+
+        # @version <= 0.4.0-prerelease
+        class Bar; end
+
+        # @version = 0.4.0
+        class Baz; end
+
+        # @version >= 0.4.0-prerelease
+        class Buzz; end
+      RBI
+
+      tree = Parser.parse_string(rbi)
+
+      Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.4.0-prerelease"))
+      assert_equal(<<~RBI, tree.string)
+        # @version <= 1.0.0
+        class Foo; end
+
+        # @version <= 0.4.0-prerelease
+        class Bar; end
+
+        # @version >= 0.4.0-prerelease
+        class Buzz; end
+      RBI
+    end
+
+    def test_filter_versions_and
+      rbi = <<~RBI
+        # @version > 0.3.0, < 1.0.0
```

Comment:
Note: this implementation supports and/or logic. The line above is an example of "and" -- the version must be greater than 0.3.0 AND less than 1.0.0. Line 218 is an example of "or" logic.

Reply:
> I wonder if only permitting the same syntax as a `.gemspec` would be feasible
Could you please elaborate on this a bit more @andyw8?